server/runtime/src/main/java/org/infinispan/server/security/UserTool.java

```diff
@@ -0,0 +1,192 @@
+      UserTool userTool = new UserTool();
+      userTool.run(args);
+   }
+
+   protected void handleArgumentCommand(String command, String parameter, Iterator<String> args) {
+      switch (command) {
+         case "-b":
+         case "--batch-mode":
+            batchMode = true;
+            break;
+         case "-f":
+            parameter = args.next();
+            // Fall through
+         case "--users-file":
+            usersFileName = parameter;
+            break;
+         case "-p":
+            parameter = args.next();
+            // Fall through
+         case "--password":
+            password = parameter;
+            break;
+         case "-g":
+            parameter = args.next();
+            // Fall through
+         case "--groups":
+            for (String group : parameter.split(",")) {
+               addGroups.add(group);
+            }
+            break;
+         case "-w":
+            parameter = args.next();
+            // Fall through
+         case "--groups-file":
+            groupsFileName = parameter;
+            break;
+         case "-r":
+            parameter = args.next();
+            // Fall through
+         case "--realm":
+            realm = parameter;
+            break;
+         case "-s":
+            parameter = args.next();
+            // Fall through
+         case "--server-root":
+            serverRoot = new File(parameter);
+            break;
+         case "-u":
+            parameter = args.next();
+            // Fall through
+         case "--user":
+            username = parameter;
+            break;
+         default:
+            throw new IllegalArgumentException();
+      }
+   }
+
+   @Override
+   protected void runInternal() {
+      while (username == null || username.isEmpty()) {
+         username = System.console().readLine("Username: ");
+      }
+      File confDir = new File(serverRoot, Server.DEFAULT_SERVER_CONFIG);
+      Properties users = new Properties();
+      File usersFile = new File(confDir, usersFileName);
+      if (usersFile.exists()) {
+         try (FileReader reader = new FileReader(usersFile)) {
+            users.load(reader);
+         } catch (IOException e) {
+            throw new RuntimeException(e);
+         }
+      }
+      Properties groups = new Properties();
+      File groupsFile = new File(confDir, groupsFileName);
+      if (groupsFile.exists()) {
+         try (FileReader reader = new FileReader(groupsFile)) {
+            groups.load(reader);
+         } catch (IOException e) {
+            throw new RuntimeException(e);
+         }
+      }
+      if (password == null && !users.containsKey(username)) {
+         if (!batchMode) {
+            while (password == null || password.isEmpty()) {
+               password = new String(System.console().readPassword("Password: "));
+            }
+         } else {
+            stdErr.printf("Cannot add user '%s' without a password.\n", username);
```

Comment:
Long term shouldn't these messages come from a format that can be internationalised?